client/src/java/com/zimbra/client/ZSmartFolder.java

```diff
@@ -0,0 +1,76 @@
+package com.zimbra.client;
+
+import org.json.JSONException;
+
+import com.zimbra.client.event.ZModifySmartFolderEvent;
+import com.zimbra.common.service.ServiceException;
+import com.zimbra.common.soap.Element;
+import com.zimbra.common.soap.MailConstants;
+import com.zimbra.soap.mail.type.RetentionPolicy;
+import com.zimbra.soap.mail.type.TagInfo;
+
+public class ZSmartFolder implements Comparable<ZSmartFolder>, ZItem, ToZJSONObject {
+
+    private String id;
+    private String name;
+    private int unreadCount;
```

Comment:
I notice that ZTag also has a `mColor`.  Is there any mileage in extending `ZTag`?  Hmm, or perhaps that makes the client mirror the underlying server implementation rather than reflecting the client view.  Either way, I could imagine a re-org with both extending another class being of value.

Reply:
I tried to avoid having the client be aware that SmartFolders are just Tags, since that's an implementation detail. That said, it might be worthwhile to have some shared code there, perhaps with a `ZTagBase` or some such.